ros2/src/lanefollowingtest/SobelVidCam.py

```diff
@@ -0,0 +1,665 @@
+    
+    center_dist,lane_center_position= get_car_position(l_fit,r_fit,w,h) #getting the car distance from the center
+    return curve_radius, center_dist,width_lane,lane_center_position
+
+def undistort(img,mtx,dist):
+    return cv2.undistort(img,mtx,dist, None, mtx)
+
+
+def Lane_pipeline(img,smoothen,prevFrameCount):
+    undistorted_image= undistort(img,mtx,dist)
+    warped_image,M= warp_image(undistorted_image)
+    image_S_channel= cv2.cvtColor(warped_image, cv2.COLOR_RGB2HLS)[:,:,2]
+    
+    imgY, imgCr, imgb, imgS= Custom_channel_converter(warped_image)
+    
+    Ybinary= channelwise_thresholding(imgY,(215,255))
+    Crbinary= channelwise_thresholding(imgCr,(215,255))
+    Lbinary= channelwise_thresholding(imgb,(215,255))
+    Sbinary= channelwise_thresholding(imgS,(200,255))
+    combined = np.zeros_like(imgY)
+    
+    
+#     sobel_mag_image= sobel_mag(image_S_channel, (15,60), False)
+    # sobel_image1= sobel_image(image_S_channel,'x', 15,60, False)
+    # sobel_grad_image= sobel_gradient_image(image_S_channel,  (0.5,1.8), False)
+    # combined[(Crbinary==1)|(Ybinary==1)|((Lbinary==1)&(Sbinary==1))] = 1
+#     |((sobel_image1==1) & (sobel_grad_image==1))
+#     plt.imshow(combined)
+#     combined[]=1
+    
+#     |((sobel_image1==1)&(sobel_grad_image==1))
+#     ((sobel_mag_image == 1) & (sobel_grad_image == 0))
+    
+#     out_img,out_img1, left_fitx,right_fitx,ploty,left_curverad,right_curverad,center_dist= Plot_line(combined)
+    out_img,out_img1, left_fitx,right_fitx,ploty,left_fit, right_fit,left_lane_inds,right_lane_inds,lane_width= Plot_line(Ybinary,smoothen,prevFrameCount)
+    curverad,center_dist,width_lane,lane_center_position= calc_radius_position(combined,left_fit, right_fit,left_lane_inds,right_lane_inds,lane_width)
+    laneImage,new_img =draw_lane(img, combined, left_fitx, right_fitx, M)
+    unwarped_image= reverse_warping(laneImage,M)
+    laneImage = cv2.addWeighted(new_img, 1, unwarped_image, 0.5, 0)
+    laneImage, copy = Plot_details(laneImage,curverad,center_dist,width_lane,lane_center_position)
+    # print(center_dist)
+    return img,out_img,out_img1,unwarped_image,laneImage,combined,copy
+
+    
+
+def CallPipeline(image):
+    smoothen= True
+    prevFrameCount=4
+    rgb_image,out_img,out_img1,unwarped_image,laneImage,combined,data_copy= Lane_pipeline(image,smoothen,prevFrameCount)
+
+    out_image = np.zeros((720,1280,3), dtype=np.uint8)
+    
+    #stacking up various images in one output Image
+    out_image[0:720,0:1280,:] = cv2.resize(laneImage,(1280,720)) #top-left
+    out_image[20:190,960:1260,:] = cv2.resize(np.dstack((combined*255, combined*255, combined*255)),(300,170))#side Panel
+    out_image[210:380,960:1260,:] = cv2.resize(out_img,(300,170))#side Panel
+#     out_image[400:570,960:1260,:] = cv2.resize(data_copy,(300,170))#bottom-left
+    return out_image
+
+
+       
+def main():
+    # Set parameters
+    cap = cv2.VideoCapture("/dev/video2")
```

Comment:
Please use `argparse` to have this, along with all hard-coded paths to be parameters that are passed at runtime.